kernel/src/table_changes/log_replay/tests.rs

```diff
@@ -228,41 +212,68 @@ async fn incompatible_schemas_fail() {
     }
 
     // The CDF schema has fields: `id: int` and `value: string`.
-    // This commit has schema with fields: `id: long` and `value: string`.
+    // This commit has schema with fields: `id: long`, `value: string` and `year: int` (non-nullable).
     let schema = StructType::new([
         StructField::new("id", DataType::LONG, true),
         StructField::new("value", DataType::STRING, true),
+        StructField::new("year", DataType::INTEGER, false),
     ]);
-    assert_incompatible_schema(schema).await;
+    assert_incompatible_schema(schema, get_schema()).await;
+
+    // The CDF schema has fields: `id: int` and `value: string`.
+    // This commit has schema with fields: `id: long`, `value: string` and `year: int` (nullable).
+    let schema = StructType::new([
+        StructField::new("id", DataType::LONG, true),
+        StructField::new("value", DataType::STRING, true),
+        StructField::new("year", DataType::INTEGER, true),
+    ]);
```

Comment:
I'm not sure this is a very helpful test. It is probably the same test as the previous one in practice, because the check fails at the existence of an unexpected field, without even checking its nullability.